kibicara/kibicara_mda.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2020 by Maike <maike@systemli.org>
+#
+# SPDX-License-Identifier: 0BSD
+
+import email.parser
+from email.policy import default
+import email.message
+import sys
+import re
+import requests
+from logging import getLogger
+from kibicara.model import Hood
+from kibicara.platforms.email.model import Email
+import argparse
+from asyncio import run
+from fastapi import status
+from ormantic import NoMatch
+
+
+async def async_main(mail=None, hood_name=None):
+    logger = getLogger(__name__)
+
+    # the MDA passes the recipient address as command line argument
+    parser = argparse.ArgumentParser()
+    if hood_name is None:
+        parser.add_argument("recipient_address")
+        args = parser.parse_args()
+        # extract hood name from the envelope recipient address
+        hood_name = args.recipient_address.split('@')[0].lower()
+        if hood_name.startswith('kibicara.'):
+            hood_name = hood_name[9:]
+        else:
+            logger.error("Recipient address didn't start with 'kibicara.'")
+
+    if mail is None:
+        # read mail from STDIN
+        mail = bytes(sys.stdin.read(), encoding='ascii')
```

Comment:
use `sys.stdin.buffer.read()` to read `bytes` and support emails which use non-ascii without base64-encoding.